File: JS/51-60/e_56.js
function divMult(n, a, numUmbral) {
    let numerosEncontrados = [];
    for (let num = 0; num <= numUmbral; num++) {
        if (num % n === 0 && num % a !== 0) {
            numerosEncontrados.push(num);
        }
    }
    return numerosEncontrados;
}

// Ejemplo de uso:
const n = 3;
const a = 4;
const numUmbral = 20;
const resultado = divMult(n, a, numUmbral);
console.log(`Números divisibles por ${n} y no múltiplos de ${a} hasta ${numUmbral}:`, resultado);


// Esta función recorre cada número dentro del rango desde 0 hasta numUmbral. Para cada número, verifica si es divisible por n y no es múltiplo de a. 
// Si cumple ambas condiciones, se agrega el número a la lista numerosEncontrados. Finalmente, se devuelve la lista de números encontrados.

File: Python/51-60/e_56.py
# Escribe una función en Python llamada divMult(n,a,numUmbral) que permita encontrar los números (desde 0 hasta numUmbral)
# que son divisibles por n y no son múltiplos de a.

def divMult(n, a, numUmbral):
    numeros_encontrados = []
    for num in range(numUmbral + 1):
        if num % n == 0 and num % a != 0:
            numeros_encontrados.append(num)
    return numeros_encontrados

# Ejemplo de uso:
n = 3
a = 4
numUmbral = 20
resultado = divMult(n, a, numUmbral)
print(f"Números divisibles por {n} y no múltiplos de {a} hasta {numUmbral}: {resultado}")


# Esta función recorre cada número dentro del rango desde 0 hasta numUmbral. Para cada número, verifica si es divisible por n y no es múltiplo de a. 
# Si cumple ambas condiciones, agrega el número a la lista de numeros_encontrados. Finalmente, devuelve la lista de números encontrados.